server/repository/src/db_diesel/temperature_chart/temperature_chart_row.rs

```diff
@@ -0,0 +1,224 @@
+use crate::{db_diesel::temperature_log_row::temperature_log, DBType};
+use chrono::NaiveDateTime;
+use diesel::{
+    backend::Backend, expression::*, prelude::*, query_builder::*, query_source::*, sql_types::*,
+};
+
+#[derive(Debug, PartialEq)]
+pub struct TemperatureChartRow {
+    pub from_datetime: NaiveDateTime,
+    pub to_datetime: NaiveDateTime,
+    pub average_temperature: f64,
```

Comment:
I had it like this originally, but ran into problems when doing filtering, since temperature log filter is in the overall filter clause (rather then a filter in join), intervals will be missing anyways if temperature logs are missing. Intervals are returned by the repository, and if you have a main axis with intervals (with say no data), and then other axis for sensors/logs with missing data, it should work correctly.

I added this: 73179598e55dfad83e544dfe9de31b6b944d7eb1 to README.md